File: 169-majority-element/majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) 
    {
        int lastnum = nums[0];
        int count =1;
        for(int i=1; i<nums.size(); i++)
        {
            if(nums[i] == lastnum) count++;
            else count--;

            if(count == 0)  {
                lastnum = nums[i];
                count=1;
            }
        }
        return lastnum;
    }
};

File: 169-majority-element/majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        count = 1
        i = 1
        nums.sort()
        while i<len(nums):
            if nums[i] == nums[i-1]:
                count+=1
            else:
                count=1
            if count > len(nums)/2:
                return nums[i]
            i+=1
        return nums[0]